modules/signals/src/state-source.ts

```diff
@@ -29,6 +30,17 @@ export type StateWatcher<State extends object> = (
   state: NoInfer<State>
 ) => void;
 
+export function isWritableStateSource<State extends object>(
+  stateSource: StateSource<State>
+): stateSource is WritableStateSource<State> {
+  return (
```

Comment:
Why not [isWritableSignal](https://github.com/angular/angular/blob/main/packages/core/src/render3/reactivity/signal.ts#L107C17-L107C33)?

Reply:
Because `isWritableSignal` is not part of the public API: https://angular.dev/api/core/isWritableSignal